python_modules/dagster/dagster_tests/storage_tests/branching_io_manager_tests/utils.py

```diff
@@ -0,0 +1,107 @@
+from contextlib import contextmanager
+from typing import Any, List, Optional, Sequence, Union
+
+from dagster import (
+    AssetKey,
+    DagsterInstance,
+    Definitions,
+    ExecuteInProcessResult,
+    IOManager,
+    InputContext,
+    OutputContext,
+)
+from dagster._core.definitions.events import CoercibleToAssetKey
+from dagster._core.event_api import EventLogRecord, EventRecordsFilter
+from dagster._core.events import DagsterEventType
+
+
+class DefinitionsRunner:
```

Comment:
We chatted about this briefly on Zoom, and I don't think it needs to block this PR, but I think this is functionality that many users would find useful, and we should build a public-facing version of it for use in both user tests and notebooks.

Here's a version of this from pre-Definitions world: https://github.com/dagster-io/dagster/pull/10072. To support the case when the `Definitions` has a non-in-process executor, it requires the `Definitions` to be defined at module-scope, but we could relax that constraint when the executor is in-process.